src/schema/city/index.ts

```diff
@@ -100,13 +104,29 @@ export const City = {
   description: "A city-based entry point for local discovery",
   args: {
     slug: {
-      type: new GraphQLNonNull(GraphQLString),
+      type: GraphQLString,
       description:
         "A slug for the city, conforming to Gravity's city slug naming conventions",
     },
+    near: {
+      type: Near,
+      description:
+        "A point which will be used to locate the nearest local discovery city within a threshold",
+    },
   },
-  resolve: (_obj, args) => {
-    return lookupCity(args.slug)
+  resolve: (_obj, { slug, near }) => {
+    if (slug && near) {
+      throw new Error('The "slug" and "near" arguments are mutually exclusive.')
+    }
+    if (!slug && !near) {
+      throw new Error('One of the arguments "slug" or "near" is required.')
+    }
+    if (slug) {
+      return lookupCity(slug)
+    }
+    if (near) {
+      return nearestCityOrNull(near)
```

Comment:
Is returning a-value-or-null idiomatic?

Reply:
Sure. Unless you mean _naming_ it like that? In which case, I don’t believe so.

Reply:
Cool, I'll just simplify the name then.